indexer/feature.hpp

```diff
@@ -142,7 +142,7 @@ class FeatureType
   }
   //@}
 
-  std::string DebugString(int scale, bool includeKeyPoint = true);
+  std::string DebugString(int scale = FeatureType::BEST_GEOMETRY, bool includeKeyPoint = true);
```

Comment:
Is this default parameter really necessary? Isn't it clearer with the explicit syntax?

Reply:
Its almost never needed, because usually geometry is loaded already.
And if its not loaded then doesn't matter much which one to load as its used solely to display coords so that its easier to locate the feature.

In some specific cases it could make a difference (e.g. we want to just log and then continue working with the feature), so I didn't remove it completely.

Reply:
Isn't loading the best geometry slower? I would prefer an explicit specialization in the place where it is used. It makes the author's intention and needs clearer.

Reply:
I could change to WORST_GEOMETRY of course, but it doesn't really matter for firing an assert.
And where it matters an explicit zoom param could be passed still.

Reply:
I'm against any default parameter here. So the function's user will always be forced to explicitly specify what is needed.

Reply:
Default value is ok here. A developer just wants to print a Feature for debug purpose. Why we do force him to think about scales ..

Reply:
Then let's remove this parameter from the function's signature.

Reply:
@biodranik do you mean your preferred approach would be overloading? e.g.
```
std::string DebugString();
std::string DebugString(int scale);
```
If so, why is it better?

Reply:
I don't see a problem why having scale and its default is bad. For regular and advanced experience.